compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -516,6 +516,27 @@ impl<'a> Parser<'a> {
             token::BinOp(token::And) | token::AndAnd => {
                 make_it!(this, attrs, |this, _| this.parse_borrow_expr(lo))
             }
+            token::BinOp(token::Plus) => {
```

Comment:
This should only be done if the following token is a numeric literal. As this stands now, if you write `+foo` or `+""` this error will trigger as well. The suggestion won't be *wrong per-se*, but in those cases the problems are deeper than the `+`.

You can use `Parser::look_ahead` which takes a closure giving you a token for whatever `N` tokens you are advancing, so you can do something like `token::BinOp(token::Plus) if this.look_ahead(1, |t| t.is_numeric_literal() => {` or something. We need to decide whether an identifier like `+foo` is something where we want to also give this error or not. Personally I think we might not want to.

Reply:
I think it makes sense to limit the check to just numeric literals, made changes to reflect that.